include/knowhere/dataset.h

```diff
@@ -24,7 +24,7 @@
 
 namespace knowhere {
 
-class DataSet {
+class DataSet : public std::enable_shared_from_this<const DataSet> {
```

Comment:
There is a simpler solution for this. Just pass `std::shared_ptr<DataSet>` or `std::shared_ptr<DataSet>&` only across the code (such as function parameters) and get rid of passing `DataSet&` ones